lib/active_model/serializer.rb

```diff
@@ -26,35 +26,12 @@ class << self
     end
 
     def self.inherited(base)
-      base._attributes = self._attributes.try(:dup) || []
-      base._attributes_keys = self._attributes_keys.try(:dup) || {}
```

Comment:
I'm addressing

```ruby
self.inherited(base)
  base._attributes = self._attributes.try(:dup) || []
  base._attributes_keys = self._attributes_keys.try(:dup) || {}
end
```

Which reads to me says

- When a Serializer is subclassed it inherits a copy of `_attributes` and `_attributes_keys` from its superclass
  - It may mutate its `_attributes` and `_attributes_keys` without the changes affecting any superclasses
  - If the superclass has not set `_attributes` and `_attributes_keys`, assign them defaults of `[]` and `{}` respectively.

Since the only scenario where a Serializer is being subclassed and the `_attributes` and `_attributes_keys` not set is when `ActiveModel::Serializer` itself is being subclassed, rather than assigning defaults in the inherited hook, why not just set the defaults in the `ActiveModel::Serializer` itself? 

The other stuff I wrote about `class_attribute` and `cattr_accessor` are just about simplifying the code around that.

Does that make sense?